util/reggen/reg_block.py

```diff
@@ -77,6 +85,7 @@ def build_blocks(block: 'RegBlock',
         if isinstance(raw, list):
             # This is the simple syntax
             block.add_raw_registers(raw, 'registers field at top-level')
+            block.handle_async(None, clocks, bus)
```

Comment:
Rather than calling a "handle_async" function separately, could this code be folded into `add_raw`? It avoids having to call two methods in order to perform one action (add some registers to the block).